openedx/core/djangoapps/programs/utils.py

```diff
@@ -424,3 +432,48 @@ def _attach_run_mode_upgrade_url(self, run_mode):
                 run_mode['upgrade_url'] = None
         else:
             run_mode['upgrade_url'] = None
+
+
+def deprecated_start_datetime_text(start_date, advertised_start=None):
```

Comment:
This strikes me as a half-measure. Have you looked into using `DateUtils` in the LMS' programs JS? The stated goal of this PR is to "remove datetime-to-text functions in the LMS," but it seems like that can't be achieved without removing this function entirely. I know from experience that if a function is left in the codebase, it will continue to be used. Why not make that final leap?

If this must stay, I think this is the wrong place for this function. This module is meant to house program utilities. If we're not able to get rid of the original function, `course_start_datetime_text()` in course_metadata_utils`, I don't think there's any value in moving it.